src/transformStateWithClones.js

```diff
@@ -7,7 +7,45 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateHistory = [];
+  let currentState = { ...state };
+
+  function unexpectedActionType(message) {
+    stateHistory.push({ error: message });
+  }
+
+  for (const action of actions) {
+    let newState = { ...currentState };
+
+    switch (action.type) {
+      case 'clear':
+        newState = {};
+        break;
+
+      case 'addProperties':
+        newState = { ...newState, ...action.extraData };
+        break;
+
+      case 'removeProperties':
+        let tempState = { ...newState };
+
+        for (const key of action.keysToRemove) {
+          const { [key]: unusedVariable, ...rest } = tempState;
+
+          tempState = rest;
+        }
+        newState = tempState;
```

Comment:
You can use a delete here https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/delete